### DISCUSSION TITLE
get information of current activity

### DISCUSSION THREAD

**Discussion by liuliang-wt** [elsa-workflows/elsa-core#4426] [Q&A]
in version 2, is there a simple way to get information of the current activity of a workflow instance? especially the activity's property values.

**Comment by sfmskywalker** [comment]
Hello @liuliang-wt,

To access information about the current activity:

1. Use `workflowInstance.CurrentActivity.ActivityId` to get the current activity ID.
2. Fetch activity data with `workflowInstance.ActivityData[currentActivityId]`.

Example:

```csharp
var currentActivityId = workflowInstance.CurrentActivity.ActivityId;
var activityProperties = workflowInstance.ActivityData[currentActivityId];
var propertyValue = activityProperties["Text"];
```

Hope this helps!
